File: js/sumPairs.js
const sumPairs = (arr, num) => {

  let answerArr = [];
  let arrSort = arr.sort();

  let start = 0;
  let end = arrSort.length - 1;

  while (start < end) {
    if (arrSort[start] + arrSort[end] > num){
      end--
    }
    else if (arrSort[start] + arrSort[end] < num){
      start++
    } 
    else {
      answerArr.push([arrSort[start], arrSort[end]])
      end--
    }
  }
  //console.log(answerArr)
  if(answerArr.length){
    return answerArr
  } else {
    return 'unable to find pairs'
  }
};

module.exports = sumPairs

File: python/sum_pairs.py



def sum_pairs(list, num):
    new_list = list.copy()
    list_pairs = []

    for a in range(len(new_list) - 1):
        start = new_list[a]
        # print(new_list[a])
        for b in range(a + 1, len(new_list)):
            # print(new_list[b])
            end = new_list[b]
            if start + end == num:
                list_pairs.append([new_list[a], new_list[b]])
            
    if len(list_pairs) > 0:
         return list_pairs
    elif list_pairs == []:
       return 'unable to find pairs'
    

# sum_pairs([1,2,3,4,5], 14) # [[4,5]]
